packages/@aws-cdk/aws-ec2/lib/launch-template.ts

```diff
@@ -0,0 +1,688 @@
+   * below its desired capacity.
+   *
+   * @default One-time spot request.
+   */
+  readonly requestType?: SpotRequestType;
+
+  /**
+   * The end date of the request. For a one-time request, the request remains active until all instances
+   * launch, the request is canceled, or this date is reached. If the request is persistent, it remains
+   * active until it is canceled or this date and time is reached.
+   *
+   * @default The default end date is 7 days from the current date.
+   */
+  readonly validUntil?: Expiration;
+};
+
+/**
+ * The types of resource tags that can be set on a launch template.
+ */
+export enum LaunchTemplateTagResourceType {
+  // dev-note: Full list at https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-tagspecification.html#cfn-ec2-launchtemplate-tagspecification-resourcetype
+  // These are the only ones from the list that presently support tag on create.
+
+  /**
+   * EC2 Instances.
+   */
+  INSTANCE = 'instance',
+
+  /**
+   * EBS Volumes.
+   */
+  VOLUME = 'volume',
+}
+
+/**
+ * Interface for defining the tag specification of a LaunchTemplate.
+ */
+export interface LaunchTemplateTagSpecification {
+  /**
+   * The types of resources that are tagged with the given tags. Note that there are
+   * restrictions regarding which resources can be tagged when created. See the following
+   * for additional information.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-tagspecification.html#cfn-ec2-launchtemplate-tagspecification-resourcetype
+   */
+  readonly resourceTypes: LaunchTemplateTagResourceType[];
+
+  /**
+   * The tags to apply to the resources.
+   */
+  readonly tags: CfnTag[];
+}
+
+/**
+ * Properties of a LaunchTemplate.
+ */
+export interface LaunchTemplateProps {
+  /**
+   * Name for this launch template.
+   *
+   * @default Automatically generated name
+   */
+  readonly launchTemplateName?: string;
+
+  /**
+   * Type of instance to launch.
+   *
+   * @default - This Launch Template does not specify a default Instance Type.
+   */
+  readonly instanceType?: InstanceType;
+
+  /**
+   * The AMI that will be used by instances.
+   *
+   * @default - This Launch Template does not specify a default AMI.
+   */
+  readonly machineImage?: IMachineImage;
+
+  /**
+   * The AMI that will be used by instances.
+   *
+   * @default - This Launch Template creates a UserData based on the type of provided machineImage.
+   */
+  readonly userData?: UserData;
+
+  /**
+   * An IAM role to associate with the instance profile that is used by instances.
+   *
+   * The role must be assumable by the service principal `ec2.amazonaws.com`:
+   *
+   * @example
+   * const role = new iam.Role(this, 'MyRole', {
+   *   assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com')
+   * });
+   *
+   * @default - A new Role is created.
```

Comment:
Wait, is it though?

Shouldn't this be just as optional as all the other fields, and not create anything by default to not mess with the downstream consumers too much?